spec/controllers/sso/logins_controller_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe Sso::LoginsController do
+  describe 'POST /sso/login' do
+    context 'resource token matches sha1 of vals passed in' do
```

Comment:
You might extract these tests as units run against `ResourceTokenCreator`, and then just test that the controller collaborates with `ResourceTokenCreator` correctly.

Reply:
updated at d751de2